python/cudf/cudf/core/scalar.py

```diff
@@ -273,19 +271,19 @@ def _binop_result_dtype_or_error(self, other, op):
         return cudf.dtype(out_dtype)
 
     def _binaryop(self, other, op: str):
-        if isinstance(other, (ColumnBase, Series, BaseIndex, np.ndarray)):
-            # dispatch to column implementation
-            return NotImplemented
-        other = to_cudf_compatible_scalar(other)
-        out_dtype = self._binop_result_dtype_or_error(other, op)
-        valid = self.is_valid and (
-            isinstance(other, np.generic) or other.is_valid
-        )
-        if not valid:
-            return Scalar(None, dtype=out_dtype)
+        if is_scalar(other):
+            other = to_cudf_compatible_scalar(other)
+            out_dtype = self._binop_result_dtype_or_error(other, op)
+            valid = self.is_valid and (
+                isinstance(other, np.generic) or other.is_valid
+            )
+            if not valid:
+                return Scalar(None, dtype=out_dtype)
```

Comment:
I realize it's not in scope, so I'd be perfectly happy with a follow-up, but this seems like an important case to have tested somewhere. It's basically for any binary op between two scalars where one is NA, right? I'm surprised that's not tested, I guess because we only test column + scalar?

Reply:
(Approving but would also like to see tests for this at some point.)